pkg/csi/service/vanilla/controller.go

```diff
@@ -760,6 +757,22 @@ func (c *controller) ControllerExpandVolume(ctx context.Context, req *csi.Contro
 	log := logger.GetLogger(ctx)
 	log.Infof("ControllerExpandVolume: called with args %+v", *req)
 
+	var volumePath string
+	if strings.Contains(req.VolumeId, ".vmdk") {
+		volumePath = req.VolumeId
+	}
+	if volumePath != "" {
+		if containerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIMigration) {
+			// Migration feature switch is enabled
+			msg := fmt.Sprintf("Cannot expand migrated vSphere volume. :%q", volumePath)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Unimplemented, msg)
```

Comment:
Question: If we return Unimplemented error code, would resizer stop calling ControllerExpandVolume? Irrespective of whether CSI migration FSS is enabled or not, how would the devops make sure that he cancels the expand volume operation? Can devops just revert back the storage size to original size? I don't think Kubernetes allows to reduce the requested storage size, but please clarify.

Reply:
> If we return Unimplemented error code, would resizer stop calling ControllerExpandVolume?

No, but it should. error code doc - https://github.com/grpc/grpc-go/blob/master/codes/codes.go#L152-L160

>  how would the DevOps make sure that he cancels the expand volume operation?

once initiated deops can not cancel this operation. this is concerning. I am worried about logs filled up unnecessarily.

>  Can DevOps just revert back the storage size to the original size? I don't think Kubernetes allows to reduce the requested storage size, but please clarify

No, we can not revert size. The reduction in size is not permitted.

Reply:
@SandeepPissay @xing-yang @chethanv28  

I will be fixing k8s to block the expand volume call if the in-tree plugin is not capable of expanding volume. 
So in future k8s releases, we do not observe this issue.

follow the conversation here - https://kubernetes.slack.com/archives/CG04EL876/p1596596991041600